pkgs/applications/networking/irc/weechat/default.nix

```diff
@@ -18,64 +19,92 @@ assert tclSupport -> tcl != null;
-    ++ optional (!luaSupport)   "-DENABLE_LUA=OFF"
-    ++ optional (!perlSupport)  "-DENABLE_PERL=OFF"
-    ++ optional (!rubySupport)  "-DENABLE_RUBY=OFF"
-    ++ optional (!tclSupport)   "-DENABLE_TCL=OFF"
-    ;
+      buildInputs = with stdenv.lib; [
+          ncurses python openssl aspell gnutls zlib curl pkgconfig
+          libgcrypt pycrypto makeWrapper
+          cmake
+          asciidoctor
+          ]
+        ++ optionals stdenv.isDarwin [ pync libobjc libresolv ]
+        ++ optional  guileSupport    guile
+        ++ optional  luaSupport      lua5
+        ++ optional  perlSupport     perl
+        ++ optional  rubySupport     ruby
+        ++ optional  tclSupport      tcl
+        ++ extraBuildInputs;
 
-  buildInputs = with stdenv.lib; [
-      ncurses python openssl aspell gnutls zlib curl pkgconfig
-      libgcrypt pycrypto makeWrapper
-      cmake
-      asciidoctor
-      ]
-    ++ optionals stdenv.isDarwin [ pync libobjc libresolv ]
-    ++ optional  guileSupport    guile
-    ++ optional  luaSupport      lua5
-    ++ optional  perlSupport     perl
-    ++ optional  rubySupport     ruby
-    ++ optional  tclSupport      tcl
-    ++ extraBuildInputs;
+      NIX_CFLAGS_COMPILE = "-I${python}/include/${python.libPrefix}"
+        # Fix '_res_9_init: undefined symbol' error
+        + (stdenv.lib.optionalString stdenv.isDarwin "-DBIND_8_COMPAT=1 -lresolv");
 
-  NIX_CFLAGS_COMPILE = "-I${python}/include/${python.libPrefix}"
-    # Fix '_res_9_init: undefined symbol' error
-    + (stdenv.lib.optionalString stdenv.isDarwin "-DBIND_8_COMPAT=1 -lresolv");
+      postInstall = with stdenv.lib; ''
+        NIX_PYTHONPATH="$out/lib/${python.libPrefix}/site-packages"
+        wrapProgram "$out/bin/weechat" \
+          ${optionalString perlSupport "--prefix PATH : ${perl}/bin"} \
+          --prefix PATH : ${pythonPackages.python}/bin \
+          --prefix PYTHONPATH : "$PYTHONPATH" \
+          --prefix PYTHONPATH : "$NIX_PYTHONPATH"
+        for o in $outputs; do
+          from=$out/lib/weechat/plugins/$o.so
+          to=''${!o}/lib/weechat/plugins/$o.so
+          if [[ -e $from ]]
+          then
+            mkdir -p $(dirname $to)
+            mv $from $to
+          fi
+        done
+      '';
 
-  postInstall = with stdenv.lib; ''
-    NIX_PYTHONPATH="$out/lib/${python.libPrefix}/site-packages"
-    wrapProgram "$out/bin/weechat" \
-      ${optionalString perlSupport "--prefix PATH : ${perl}/bin"} \
-      --prefix PATH : ${pythonPackages.python}/bin \
-      --prefix PYTHONPATH : "$PYTHONPATH" \
-      --prefix PYTHONPATH : "$NIX_PYTHONPATH"
+      meta = {
+        homepage = http://www.weechat.org/;
+        description = "A fast, light and extensible chat client";
+        license = stdenv.lib.licenses.gpl3;
+        maintainers = with stdenv.lib.maintainers; [ lovek323 garbas the-kenny ];
+        platforms = stdenv.lib.platforms.unix;
+      };
+    };
+in weechat // {
```

Comment:
I'm aware of this; the idea behind the approach I've used here is to take advantage of weechat's modularity to significantly reduce weechat's closure size without the need for rebuilding weechat itself (as is necessary if any of the `xSupport` flags are passed in). Customising weechat's plugins has already been possible simply by disabling them in the build, but that means that it needs to be rebuilt on the user's end which somewhat dampens the resulting improvement in closure size as all the build deps need to be available (at least initially) and the build takes longer.